shapely/geometry/geo.py

```diff
@@ -117,7 +118,7 @@ def shape(context):
         geoms = [shape(g) for g in ob.get("geometries", [])]
         return GeometryCollection(geoms)
     else:
-        raise ValueError("Unknown geometry type: %s" % geom_type)
+        raise GeometryTypeError("Unknown geometry type: %s" % geom_type)
```

Comment:
I think this actually could make more sense as a ValueError, as this is about the *value* in the "type" key of the `__geo_interface__` dictionary (`context["type"]`)